assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/ConfigureScreensPage.tsx

```diff
@@ -0,0 +1,418 @@
+    }
+
+    return (
+      <Container className="workflow-container">
+        <div className="h3 text-white mb-5">Configure Green Line Stations</div>
+        {layout}
+      </Container>
+    );
+  };
+
+interface ConfigurePlaceCardProps {
+  place: Place;
+  existingScreens: ExistingScreens;
+  setPlacesAndScreensToUpdate: React.Dispatch<
+    React.SetStateAction<PlaceIdsAndNewScreens>
+  >;
+  handleRemoveLocation: () => void;
+}
+
+const ConfigurePlaceCard: ComponentType<ConfigurePlaceCardProps> = ({
+  place,
+  existingScreens,
+  setPlacesAndScreensToUpdate,
+  handleRemoveLocation,
+}: ConfigurePlaceCardProps) => {
+  const [existingPendingScreens, setExistingPendingScreens] = useState<{
+    [screen_id: string]: ScreenConfiguration;
+  }>({});
+  const [newScreens, setNewScreens] = useState<ScreenConfiguration[]>([]);
+
+  useEffect(() => {
+    if (!existingScreens) return;
+
+    setExistingPendingScreens(existingScreens.pending_screens);
+  }, [existingScreens]);
+
+  useEffect(() => {
+    setPlacesAndScreensToUpdate((placesAndScreens) => {
+      const screensAtPlace = placesAndScreens[place.id];
+      const newState = { ...placesAndScreens };
+      if (screensAtPlace) {
+        newState[place.id].updated_screens = existingPendingScreens;
+        newState[place.id].new_screens = newScreens;
+      } else {
+        newState[place.id] = {
+          updated_screens: existingPendingScreens,
+          new_screens: newScreens,
+        };
+      }
+      return newState;
+    });
+  }, [existingPendingScreens, newScreens]);
+
+  const getExistingLiveScreens = () => existingScreens?.live_screens ?? {};
+
+  const hasRows =
+    Object.keys(getExistingLiveScreens()).length > 0 ||
+    Object.keys(existingPendingScreens).length > 0;
+
+  return (
+    <Container className="configure-place-card p-0">
+      <Row className="header">
+        <Col className="h5 my-auto header-name">{place.name.toUpperCase()}</Col>
+        <Col className="body--medium my-auto">Station ID: {place.id}</Col>
+        <Col className="d-flex">
+          <Button
+            className="remove-location-button"
+            onClick={handleRemoveLocation}
+          >
+            Remove Location
+          </Button>
+        </Col>
+      </Row>
+      {hasRows && (
+        <Row className="screens-table-container">
+          <Table responsive borderless className="screens-table m-0">
+            <thead className="screens-table-header">
+              <tr className="body--regular">
+                <th className="screen-id">Screen ID</th>
+                <th className="direction">Direction</th>
+                <th className="platform-location">Platform Location</th>
+                <th className="status"></th>
+              </tr>
+            </thead>
+            <tbody className="screens-table-body">
+              {Object.keys(getExistingLiveScreens()).map((screenID) => {
+                const screen = getExistingLiveScreens()[screenID];
```

Comment:
You can use `Object.entries` to get both the ID and the config simultaneously, if you like:
```ts
Object.entries(existingLiveScreens).map(([screenID, screen]) => {...})
```